src/App.tsx

```diff
@@ -1,24 +1,190 @@
+import React, {
+  useState, useCallback, useEffect, useMemo,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos, postTodo, deleteTodo } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterType } from './types/FilterType';
+import { Footer } from './components/Footer/Footer';
 
-const USER_ID = 0;
+const USER_ID = 6704;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [query, setQuery] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [loading, setLoading] = useState(false);
+  const [deletedTodoId, setDeletedTodoId] = useState([0]);
+
+  const closeError = () => setTimeout(() => setErrorText(''), 3000);
+  const handleCloseButton = () => setErrorText('');
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const fetchTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorText('Unable to load todos');
+      closeError();
+    }
+  }, []);
+
+  const addTodo = useCallback(async (title: string) => {
+    try {
+      setLoading(true);
+
+      if (!title.length) {
+        setErrorText("Title can't be empty");
+        closeError();
+
+        return;
+      }
+
+      const newTodo = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setDisabledInput(true);
+      setTempTodo({ ...newTodo, id: 0 });
+
+      const createdTodo = await postTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (error) {
+      setErrorText('Unable to add todo');
+      closeError();
+
+      throw error;
+    } finally {
+      setTempTodo(null);
+      setDisabledInput(false);
+      setLoading(false);
+    }
+  }, []);
+
+  const removeTodo = async (id: number) => {
+    try {
+      setDeletedTodoId(state => [...state, id]);
+      await deleteTodo(id);
+      setTodos(todos.filter(todo => todo.id !== id));
+    } catch {
+      setErrorText('Unable to delete a todo');
+      closeError();
+    } finally {
+      setDeletedTodoId([]);
+    }
+  };
+
+  const removeCompletedTodos = async () => {
+    try {
+      const completedIds = completedTodos.map(({ id }) => removeTodo(id));
+
+      await Promise.all(completedIds);
+      setTodos(activeTodos);
+    } catch {
+      setErrorText('Unable to delete completed todos');
+      closeError();
+    } finally {
+      setDeletedTodoId([]);
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+    addTodo(query);
```

Comment:
why you need to add todo on mounted event?

Reply:
Right, thanks, it`s redundant here)